### PR TITLE
[staging] glsl: Initial GLSL ES 3.x support.

### PR DESCRIPTION
**Pull request by JohnnyonFlame**
Started discussion on issue #64 but ended up talking with @flibitijibibo over discord who told me to open a PR and wait for a review.

Thanks again!

### CODE REVIEW

File: profiles/mojoshader_profile_glsl.c [hunk 1]
```diff
@@ -485,7 +485,7 @@ static void prepend_glsl_texlod_extensions(Context *ctx)
     //  so we'll use them if available. Failing that, we'll just fallback
     //  to a regular texture2D call and hope the mipmap it chooses is close
     //  enough.
-    if (!ctx->glsl_generated_texlod_setup)
+    if (!ctx->glsl_generated_texlod_setup && !support_glsles3(ctx))
```
Comment (flibitijibibo) [inline profiles/mojoshader_profile_glsl.c:488]: Is this something we should worry about for version 100 as well?

File: profiles/mojoshader_profile_glsl.c [hunk 2]
```diff
@@ -855,12 +882,20 @@ void emit_GLSL_attribute(Context *ctx, RegisterType regtype, int regnum,
                          int flags)
 {
     // !!! FIXME: this function doesn't deal with write masks at all yet!
+    const char *qualifier_in = "varying";
```
Comment (flibitijibibo) [inline profiles/mojoshader_profile_glsl.c]: Should this be `attribute`?

File: profiles/mojoshader_profile_glsl.c [hunk 3]
```diff
@@ -1048,14 +1083,26 @@ void emit_GLSL_attribute(Context *ctx, RegisterType regtype, int regnum,
 
         if (regtype == REG_TYPE_COLOROUT)
         {
-            if (!ctx->have_multi_color_outputs)
-                usage_str = "gl_FragColor";  // maybe faster?
+            if (support_glsles3(ctx))
+            {
+                usage_str = "_gl_FragData";
```
Comment (flibitijibibo) [inline profiles/mojoshader_profile_glsl.c:1101]: Might need a quick comment saying why ES3 needs a different path

**Review by flibitijibibo — COMMENTED** [review]
Most of this seems pretty straightforward, just a few tiny bits here and there. Everything else lgtm.

File: profiles/mojoshader_profile_glsl.c (hunk 1 above)
Comment (JohnnyonFlame) [reply profiles/mojoshader_profile_glsl.c:488]: This should be fine for 100, I'm skipping this due to the syntax being deprecated for GLSL ES 3.00.

File: profiles/mojoshader_profile_glsl.c (hunk 2 above)
Comment (JohnnyonFlame) [reply profiles/mojoshader_profile_glsl.c]: Sorta - varying for fragment, attribute for vertex. It's indeed a silly mistake tho, fixed.

File: profiles/mojoshader_profile_glsl.c (hunk 3 above)
Comment (JohnnyonFlame) [reply profiles/mojoshader_profile_glsl.c:1101]: Added.
```c
            // gl_FragColor/FragData were deprecated in favor of manually
            // defining your own output variables, GLSL ES 3.x enforces the new
            // syntax, so let's use it.
```

**Comment by JohnnyonFlame** [comment]
Ooops, sorry, force pushed because something got staged by mistake.

**Review by flibitijibibo — APPROVED** [review]
Latest works for me - will likely merge this right after the 23.11 tag, so no rush on this one!

**Pull request merged**